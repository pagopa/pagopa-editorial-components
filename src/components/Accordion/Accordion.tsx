import { Box, Container, Grid, Stack, Typography } from '@mui/material';
import { AccordionItem, type AccordionItemProps } from './AccordionItem';
export interface AccordionProps {
  title: string;
  subtitle?: string;
  description?: string | JSX.Element;
  accordionItems: AccordionItemProps[];
  theme: 'light' | 'dark';

  layout?: 'left' | 'center' | 'right';
}

export const Accordion = (props: AccordionProps) => {
  const {
    title,
    subtitle,
    description,
    accordionItems,
    theme,
    layout = 'left',
  } = props;
  const isDarkTheme = theme === 'dark';
  const bgcolor = isDarkTheme ? 'primary.main' : '#FAFAFA';
  const textColor = isDarkTheme ? 'white' : 'text.primary';

  const isCenterLayout = layout === 'center';
  const textAlignment = isCenterLayout ? 'center' : 'left';

  return (
    <Box sx={{ py: { xs: 4, md: 10 } }} bgcolor={bgcolor} component="section">
      <Container>
        <Grid spacing={{ xs: 5, md: isCenterLayout ? 7 : 16 }} container>
          <Grid item xs={12} md={isCenterLayout ? 12 : 4}>
            <Stack px={{ xs: 1, md: 0 }} spacing={{ xs: 2, md: 5 }}>
              {/** Title */}
              {title && (
                <Typography
                  textAlign={textAlignment}
                  color={textColor}
                  variant="h4"
                >
                  {title}
                </Typography>
              )}

              {/** Subtitle */}
              {subtitle && (
                <>
                  <Typography
                    textAlign={textAlignment}
                    color={textColor}
                    variant="h6"
                  >
                    {subtitle}
                  </Typography>
                </>
              )}

              {/** Description */}
              {description && (
                <>
                  {typeof description === 'string' ? (
                    <Typography
                      textAlign={textAlignment}
                      color={textColor}
                      variant="body2"
                    >
                      {description}
                    </Typography>
                  ) : (
                    <Box textAlign={textAlignment} color={textColor}>
                      {description}
                    </Box>
                  )}
                </>
              )}
            </Stack>
          </Grid>
          <Grid
            item
            order={layout === 'right' ? -1 : 1}
            xs={12}
            md={isCenterLayout ? 12 : 8}
          >
            {/** Accordions */}
            <Stack spacing={2}>
              {accordionItems.map((accordionItem, i) => (
                <AccordionItem key={i} {...accordionItem} />
              ))}
            </Stack>
          </Grid>
        </Grid>
      </Container>
    </Box>
  );
};
