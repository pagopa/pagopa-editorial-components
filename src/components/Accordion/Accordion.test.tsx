import { Accordion } from './Accordion';
import '@testing-library/jest-dom';
/**
 * @jest-environment jsdom
 */

import renderer from 'react-test-renderer';

it('renders correctly with all props', () => {
  const tree = renderer
    .create(
      <Accordion
        title="titleTest"
        subtitle="subtitleTest"
        description="descriptionTest"
        accordionItems={[{ header: 'tetsHead', content: 'testCont' }]}
        theme="light"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with all props and accordion header jsx', () => {
  const tree = renderer
    .create(
      <Accordion
        title="titleTest"
        subtitle="subtitleTest"
        description="descriptionTest"
        accordionItems={[{ header: 'tetsHead', content: <div>test</div> }]}
        theme="light"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with all props and two accordion items', () => {
  const tree = renderer
    .create(
      <Accordion
        title="titleTest"
        subtitle="subtitleTest"
        description="descriptionTest"
        accordionItems={[
          { header: 'tetsHead', content: 'testCont' },
          { header: 'tetsHead2', content: 'testCont2' },
        ]}
        theme="light"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
