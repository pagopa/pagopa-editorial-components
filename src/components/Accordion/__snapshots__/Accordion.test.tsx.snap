// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with all props 1`] = `
<section
  className="MuiBox-root css-54dja0"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5 MuiGrid-spacing-md-16 css-9mbnn2-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-1lj5egr-MuiGrid-root"
      >
        <div
          className="MuiStack-root css-w1wlmt-MuiStack-root"
        >
          <h4
            className="MuiTypography-root MuiTypography-h4 css-fokda-MuiTypography-root"
          >
            titleTest
          </h4>
          <h6
            className="MuiTypography-root MuiTypography-h6 css-1uiqjqz-MuiTypography-root"
          >
            subtitleTest
          </h6>
          <p
            className="MuiTypography-root MuiTypography-body2 css-4zcgo1-MuiTypography-root"
          >
            descriptionTest
          </p>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8 css-1nvq36h-MuiGrid-root"
      >
        <div
          className="MuiStack-root css-1p5q5e5-MuiStack-root"
        >
          <div
            className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded css-1byhwgv-MuiPaper-root-MuiAccordion-root"
          >
            <div
              aria-controls=":r0:-controls"
              aria-expanded={false}
              className="MuiButtonBase-root MuiAccordionSummary-root css-1pvvkxv-MuiButtonBase-root-MuiAccordionSummary-root"
              id=":r1:-header"
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                className="MuiAccordionSummary-content css-1betqn-MuiAccordionSummary-content"
              >
                <p
                  className="MuiTypography-root MuiTypography-body1 css-1yuoiht-MuiTypography-root"
                >
                  tetsHead
                </p>
              </div>
              <div
                className="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ExpandMoreIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </svg>
              </div>
            </div>
            <div
              className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
              style={
                {
                  "minHeight": "0px",
                }
              }
            >
              <div
                className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
              >
                <div
                  className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
                >
                  <div
                    aria-labelledby=":r1:-header"
                    className="MuiAccordion-region"
                    id=":r0:-controls"
                    role="region"
                  >
                    <div
                      className="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
                    >
                      <p
                        className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                      >
                        testCont
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`renders correctly with all props and accordion header jsx 1`] = `
<section
  className="MuiBox-root css-54dja0"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5 MuiGrid-spacing-md-16 css-9mbnn2-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-1lj5egr-MuiGrid-root"
      >
        <div
          className="MuiStack-root css-w1wlmt-MuiStack-root"
        >
          <h4
            className="MuiTypography-root MuiTypography-h4 css-fokda-MuiTypography-root"
          >
            titleTest
          </h4>
          <h6
            className="MuiTypography-root MuiTypography-h6 css-1uiqjqz-MuiTypography-root"
          >
            subtitleTest
          </h6>
          <p
            className="MuiTypography-root MuiTypography-body2 css-4zcgo1-MuiTypography-root"
          >
            descriptionTest
          </p>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8 css-1nvq36h-MuiGrid-root"
      >
        <div
          className="MuiStack-root css-1p5q5e5-MuiStack-root"
        >
          <div
            className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded css-1byhwgv-MuiPaper-root-MuiAccordion-root"
          >
            <div
              aria-controls=":r2:-controls"
              aria-expanded={false}
              className="MuiButtonBase-root MuiAccordionSummary-root css-1pvvkxv-MuiButtonBase-root-MuiAccordionSummary-root"
              id=":r3:-header"
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                className="MuiAccordionSummary-content css-1betqn-MuiAccordionSummary-content"
              >
                <p
                  className="MuiTypography-root MuiTypography-body1 css-1yuoiht-MuiTypography-root"
                >
                  tetsHead
                </p>
              </div>
              <div
                className="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ExpandMoreIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </svg>
              </div>
            </div>
            <div
              className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
              style={
                {
                  "minHeight": "0px",
                }
              }
            >
              <div
                className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
              >
                <div
                  className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
                >
                  <div
                    aria-labelledby=":r3:-header"
                    className="MuiAccordion-region"
                    id=":r2:-controls"
                    role="region"
                  >
                    <div
                      className="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
                    >
                      <div>
                        test
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`renders correctly with all props and accordion header jsx dark 1`] = `
<section
  className="MuiBox-root css-1uxhjz"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5 MuiGrid-spacing-md-16 css-9mbnn2-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-1lj5egr-MuiGrid-root"
      >
        <div
          className="MuiStack-root css-w1wlmt-MuiStack-root"
        >
          <h4
            className="MuiTypography-root MuiTypography-h4 css-1f5ya6j-MuiTypography-root"
          >
            titleTest
          </h4>
          <h6
            className="MuiTypography-root MuiTypography-h6 css-hcdnj8-MuiTypography-root"
          >
            subtitleTest
          </h6>
          <p
            className="MuiTypography-root MuiTypography-body2 css-22y7ta-MuiTypography-root"
          >
            descriptionTest
          </p>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8 css-1nvq36h-MuiGrid-root"
      >
        <div
          className="MuiStack-root css-1p5q5e5-MuiStack-root"
        >
          <div
            className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded css-1byhwgv-MuiPaper-root-MuiAccordion-root"
          >
            <div
              aria-controls=":ra:-controls"
              aria-expanded={false}
              className="MuiButtonBase-root MuiAccordionSummary-root css-1pvvkxv-MuiButtonBase-root-MuiAccordionSummary-root"
              id=":rb:-header"
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                className="MuiAccordionSummary-content css-1betqn-MuiAccordionSummary-content"
              >
                <p
                  className="MuiTypography-root MuiTypography-body1 css-1yuoiht-MuiTypography-root"
                >
                  tetsHead
                </p>
              </div>
              <div
                className="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ExpandMoreIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </svg>
              </div>
            </div>
            <div
              className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
              style={
                {
                  "minHeight": "0px",
                }
              }
            >
              <div
                className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
              >
                <div
                  className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
                >
                  <div
                    aria-labelledby=":rb:-header"
                    className="MuiAccordion-region"
                    id=":ra:-controls"
                    role="region"
                  >
                    <div
                      className="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
                    >
                      <div>
                        test
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`renders correctly with all props and two accordion items 1`] = `
<section
  className="MuiBox-root css-54dja0"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5 MuiGrid-spacing-md-16 css-9mbnn2-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-1lj5egr-MuiGrid-root"
      >
        <div
          className="MuiStack-root css-w1wlmt-MuiStack-root"
        >
          <h4
            className="MuiTypography-root MuiTypography-h4 css-fokda-MuiTypography-root"
          >
            titleTest
          </h4>
          <h6
            className="MuiTypography-root MuiTypography-h6 css-1uiqjqz-MuiTypography-root"
          >
            subtitleTest
          </h6>
          <p
            className="MuiTypography-root MuiTypography-body2 css-4zcgo1-MuiTypography-root"
          >
            descriptionTest
          </p>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8 css-1nvq36h-MuiGrid-root"
      >
        <div
          className="MuiStack-root css-1p5q5e5-MuiStack-root"
        >
          <div
            className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded css-1byhwgv-MuiPaper-root-MuiAccordion-root"
          >
            <div
              aria-controls=":r4:-controls"
              aria-expanded={false}
              className="MuiButtonBase-root MuiAccordionSummary-root css-1pvvkxv-MuiButtonBase-root-MuiAccordionSummary-root"
              id=":r5:-header"
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                className="MuiAccordionSummary-content css-1betqn-MuiAccordionSummary-content"
              >
                <p
                  className="MuiTypography-root MuiTypography-body1 css-1yuoiht-MuiTypography-root"
                >
                  tetsHead
                </p>
              </div>
              <div
                className="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ExpandMoreIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </svg>
              </div>
            </div>
            <div
              className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
              style={
                {
                  "minHeight": "0px",
                }
              }
            >
              <div
                className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
              >
                <div
                  className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
                >
                  <div
                    aria-labelledby=":r5:-header"
                    className="MuiAccordion-region"
                    id=":r4:-controls"
                    role="region"
                  >
                    <div
                      className="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
                    >
                      <p
                        className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                      >
                        testCont
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded css-1byhwgv-MuiPaper-root-MuiAccordion-root"
          >
            <div
              aria-controls=":r6:-controls"
              aria-expanded={false}
              className="MuiButtonBase-root MuiAccordionSummary-root css-1pvvkxv-MuiButtonBase-root-MuiAccordionSummary-root"
              id=":r7:-header"
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                className="MuiAccordionSummary-content css-1betqn-MuiAccordionSummary-content"
              >
                <p
                  className="MuiTypography-root MuiTypography-body1 css-1yuoiht-MuiTypography-root"
                >
                  tetsHead2
                </p>
              </div>
              <div
                className="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ExpandMoreIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </svg>
              </div>
            </div>
            <div
              className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
              style={
                {
                  "minHeight": "0px",
                }
              }
            >
              <div
                className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
              >
                <div
                  className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
                >
                  <div
                    aria-labelledby=":r7:-header"
                    className="MuiAccordion-region"
                    id=":r6:-controls"
                    role="region"
                  >
                    <div
                      className="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
                    >
                      <p
                        className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                      >
                        testCont2
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`renders correctly with all props and two accordion items dark 1`] = `
<section
  className="MuiBox-root css-1uxhjz"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5 MuiGrid-spacing-md-16 css-9mbnn2-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-1lj5egr-MuiGrid-root"
      >
        <div
          className="MuiStack-root css-w1wlmt-MuiStack-root"
        >
          <h4
            className="MuiTypography-root MuiTypography-h4 css-1f5ya6j-MuiTypography-root"
          >
            titleTest
          </h4>
          <h6
            className="MuiTypography-root MuiTypography-h6 css-hcdnj8-MuiTypography-root"
          >
            subtitleTest
          </h6>
          <p
            className="MuiTypography-root MuiTypography-body2 css-22y7ta-MuiTypography-root"
          >
            descriptionTest
          </p>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8 css-1nvq36h-MuiGrid-root"
      >
        <div
          className="MuiStack-root css-1p5q5e5-MuiStack-root"
        >
          <div
            className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded css-1byhwgv-MuiPaper-root-MuiAccordion-root"
          >
            <div
              aria-controls=":rc:-controls"
              aria-expanded={false}
              className="MuiButtonBase-root MuiAccordionSummary-root css-1pvvkxv-MuiButtonBase-root-MuiAccordionSummary-root"
              id=":rd:-header"
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                className="MuiAccordionSummary-content css-1betqn-MuiAccordionSummary-content"
              >
                <p
                  className="MuiTypography-root MuiTypography-body1 css-1yuoiht-MuiTypography-root"
                >
                  tetsHead
                </p>
              </div>
              <div
                className="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ExpandMoreIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </svg>
              </div>
            </div>
            <div
              className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
              style={
                {
                  "minHeight": "0px",
                }
              }
            >
              <div
                className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
              >
                <div
                  className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
                >
                  <div
                    aria-labelledby=":rd:-header"
                    className="MuiAccordion-region"
                    id=":rc:-controls"
                    role="region"
                  >
                    <div
                      className="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
                    >
                      <p
                        className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                      >
                        testCont
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded css-1byhwgv-MuiPaper-root-MuiAccordion-root"
          >
            <div
              aria-controls=":re:-controls"
              aria-expanded={false}
              className="MuiButtonBase-root MuiAccordionSummary-root css-1pvvkxv-MuiButtonBase-root-MuiAccordionSummary-root"
              id=":rf:-header"
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                className="MuiAccordionSummary-content css-1betqn-MuiAccordionSummary-content"
              >
                <p
                  className="MuiTypography-root MuiTypography-body1 css-1yuoiht-MuiTypography-root"
                >
                  tetsHead2
                </p>
              </div>
              <div
                className="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ExpandMoreIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </svg>
              </div>
            </div>
            <div
              className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
              style={
                {
                  "minHeight": "0px",
                }
              }
            >
              <div
                className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
              >
                <div
                  className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
                >
                  <div
                    aria-labelledby=":rf:-header"
                    className="MuiAccordion-region"
                    id=":re:-controls"
                    role="region"
                  >
                    <div
                      className="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
                    >
                      <p
                        className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                      >
                        testCont2
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`renders correctly with all props dark 1`] = `
<section
  className="MuiBox-root css-1uxhjz"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5 MuiGrid-spacing-md-16 css-9mbnn2-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-1lj5egr-MuiGrid-root"
      >
        <div
          className="MuiStack-root css-w1wlmt-MuiStack-root"
        >
          <h4
            className="MuiTypography-root MuiTypography-h4 css-1f5ya6j-MuiTypography-root"
          >
            titleTest
          </h4>
          <h6
            className="MuiTypography-root MuiTypography-h6 css-hcdnj8-MuiTypography-root"
          >
            subtitleTest
          </h6>
          <p
            className="MuiTypography-root MuiTypography-body2 css-22y7ta-MuiTypography-root"
          >
            descriptionTest
          </p>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8 css-1nvq36h-MuiGrid-root"
      >
        <div
          className="MuiStack-root css-1p5q5e5-MuiStack-root"
        >
          <div
            className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded css-1byhwgv-MuiPaper-root-MuiAccordion-root"
          >
            <div
              aria-controls=":r8:-controls"
              aria-expanded={false}
              className="MuiButtonBase-root MuiAccordionSummary-root css-1pvvkxv-MuiButtonBase-root-MuiAccordionSummary-root"
              id=":r9:-header"
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                className="MuiAccordionSummary-content css-1betqn-MuiAccordionSummary-content"
              >
                <p
                  className="MuiTypography-root MuiTypography-body1 css-1yuoiht-MuiTypography-root"
                >
                  tetsHead
                </p>
              </div>
              <div
                className="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ExpandMoreIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </svg>
              </div>
            </div>
            <div
              className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
              style={
                {
                  "minHeight": "0px",
                }
              }
            >
              <div
                className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
              >
                <div
                  className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
                >
                  <div
                    aria-labelledby=":r9:-header"
                    className="MuiAccordion-region"
                    id=":r8:-controls"
                    role="region"
                  >
                    <div
                      className="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
                    >
                      <p
                        className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                      >
                        testCont
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
